src/SimpleWeightedGraphs.jl

```diff
@@ -59,14 +59,11 @@ convert(::Type{SparseMatrixCSC{T, U}}, g::AbstractSimpleWeightedGraph) where T<:
 
 ### INTERFACE
 
-nv(g::AbstractSimpleWeightedGraph{T, U}) where T where U = T(size(g.weights, 1))
+nv(g::AbstractSimpleWeightedGraph{T, U}) where T where U = T(size(weights(g), 1))
 vertices(g::AbstractSimpleWeightedGraph{T, U}) where T where U = one(T):nv(g)
 eltype(x::AbstractSimpleWeightedGraph{T, U}) where T where U = T
 weighttype(x::AbstractSimpleWeightedGraph{T, U}) where T where U = U
 
-has_edge(g::AbstractSimpleWeightedGraph{T, U}, e::AbstractSimpleWeightedEdge) where T where U =
-    g.weights[dst(e), src(e)] != zero(U)
```

Comment:
Don't we need a fallback anymore for custom subtypes of `AbstractSimpleWeightedGraph` - or is the idea, that from now on, custom implementations need to implement their own `has_edge` function?

Reply:
Interestingly we never uses the weight of `e` here.

Reply:
I also wonder whether users need to implement both `has_edge(g, s, d)` and `has_edge(g, e)` (see https://github.com/JuliaGraphs/SimpleWeightedGraphs.jl/pull/14/files#r1005617050)

Reply:
Sorry for the delay. the automatic fallback is from the version `has_edge(g, s, t)` to `has_edge(g, e)`, but here it looks more optimal to do the opposite, otherwise `has_edge(g, s, t)` would create an edge to destructure it immediately. Should we reflect this in Graphs.jl, do the fallback every time it is necessary or just keep the slightly unoptimized version ?

Reply:
I would personally fall back on `has_edge(g, s, t)`, mostly because we don't always know what kind of edge is "the standard kind" for a given graph. But we can start by switching it here, and then maybe reflect it in Graphs.jl?

Reply:
Let's merge this for now, it's not an emergency anyway